packages/webapp/src/pages/delegates/index.tsx

```diff
@@ -22,80 +33,128 @@ interface Props {
 const MEMBERS_PAGE_SIZE = 18;
 
 export const DelegatesPage = (props: Props) => {
-    const { isError: isMembersDataFetchError, data: members } = useQuery({
+    const [ualAccount] = useUALAccount();
+    const accountName = ualAccount?.accountName;
+    const { data: currentMember } = useCurrentMember();
+
+    const { data: members } = useQuery({
         ...queryMembers(1, MEMBERS_PAGE_SIZE),
```

Comment:
I believe you can use the `queryMemberData` query here instead (passing in the `accountName`). Then a lot of this becomes quite a bit simpler. You already have the current member's data. You can display their data at the bottom since you're only displaying 1 record there anyway. And if I'm reading this right, you won't even need to invoke your `getMemberRecordFromName` function below either, because you'll already have the `MemberData`.

Also, your queries here don't need the `keepPreviousData` flag since you're not doing any server-side querying. You can get rid of those and the `getServerSideProps` function too.